presto-cache/src/main/java/com/facebook/presto/cache/LocalRangeCacheManager.java

```diff
@@ -345,6 +344,35 @@ private boolean write(FileReadRequest key, byte[] data, Path newFilePath)
         return true;
     }
 
+    private int copyFromFileInChunks(File newFile, LocalCacheFile sourceFile, long startPos)
+            throws IOException
+    {
+        int totalBytesRead = 0;
+        try (FileInputStream fs = new FileInputStream(new File(sourceFile.getPath().toUri()))) {
+            fs.getChannel().position(startPos);
+            int readBytes = 0;
+            byte[] buffer = new byte[inMemoryChunkSize];
+
+            while ((readBytes = fs.read(buffer)) != -1) {
+                totalBytesRead += readBytes;
+
+                if (readBytes != inMemoryChunkSize) {
+                    byte[] tempBuffer = new byte[readBytes];
```

Comment:
Can we avoid creating extra buffers?

Reply:
changed to using a ```RandomAccessFile``` instead, which avoids the extra buffer.